### PR TITLE
Feature #13573 : See also: improve display order of the publications

### PR DESCRIPTION
**Pull request by SilverDav**
- Publications are displayed by alphabetical order (by default)
- The order can be change by the user with the selectbox

### CODE REVIEW

File: kmelia/kmelia-war/src/main/webapp/kmelia/jsp/publicationLinksManager.jsp [hunk 1]
```diff
@@ -26,169 +26,187 @@
...
+
+  <script type="text/javascript">
+    var context ='<%=m_context %>';
+    var currentNodeId;
+    var currentNodeIndex;
+    var currentComponent;
+    var pubId = '<%=pubId%>';
+    var currentSort = <%=KmeliaPublicationSort.SORT_TITLE_ASC%>;
+    var menuType='<%=MenuConstants.SEE_ALSO_MENU_TYPE%>';
+    function buildTree() {
```
Comment (mmoqui) [inline kmelia/kmelia-war/src/main/webapp/kmelia/jsp/publicationLinksManager.jsp:73]: All the var here can be replaced either by let or by const (for the constant). This is a strong recommendation of the Javascript rules

**Review by mmoqui — CHANGES_REQUESTED** [review]

File: kmelia/kmelia-war/src/main/webapp/kmelia/jsp/publicationLinksManager.jsp (hunk 1 above)
Comment (SilverDav) [reply kmelia/kmelia-war/src/main/webapp/kmelia/jsp/publicationLinksManager.jsp:73]: Done.

File: kmelia/kmelia-war/src/main/webapp/kmelia/jsp/publicationLinksManager.jsp [hunk 2]
```diff
@@ -26,169 +26,187 @@
...
+      //create a new tree:
+      tree = new YAHOO.widget.TreeView("treeDiv1");
+      //turn dynamic loading on for entire tree:
+      tree.setDynamicLoad(loadNodeData, 0);
+      //get root node for tree:
+      var root = tree.getRoot();
+
+      //add child nodes for tree; our top level nodes are
+      try{
+        var mes = [];
```
[the remark this answers is not in the document]
Comment (SilverDav) [reply kmelia/kmelia-war/src/main/webapp/kmelia/jsp/publicationLinksManager.jsp]: Done.

File: kmelia/kmelia-war/src/main/webapp/kmelia/jsp/publicationLinksManager.jsp [hunk 3]
```diff
@@ -26,169 +26,187 @@
...
+        mes =YAHOO.lang.JSON.stringify(<%=TreeHandler.ProcessMenu(request,MenuConstants.SEE_ALSO_MENU_TYPE, true)%>);
+        mes = YAHOO.lang.JSON.parse(mes);
+      }catch(x){
+        notyError("JSON Parse failed: "+x);
+        return;
+      }
+      for (var i=0, j=mes.length; i<j; i++) {
+        var tempNode = new YAHOO.widget.TextNode(mes[i], root, false);
```
[the remark this answers is not in the document]
Comment (SilverDav) [reply kmelia/kmelia-war/src/main/webapp/kmelia/jsp/publicationLinksManager.jsp]: Done.

File: kmelia/kmelia-war/src/main/webapp/kmelia/jsp/publicationLinksManager.jsp [hunk 4]
```diff
@@ -26,169 +26,187 @@
...
+    function getCurrentNodeId(){
+      return currentNodeId;
+    }
+
+    function setCurrentNodeId(id){
+      currentNodeId = id;
+    }
+
+    function doPagination(index, nbItemsPerPage){
+      var ieFix = new Date().getTime();
```
[the remark this answers is not in the document]
Comment (SilverDav) [reply kmelia/kmelia-war/src/main/webapp/kmelia/jsp/publicationLinksManager.jsp]: Done.

File: kmelia/kmelia-war/src/main/webapp/kmelia/jsp/publicationLinksManager.jsp [hunk 5]
```diff
@@ -26,169 +26,187 @@
...
+          },"html");
+    }
+
+    function closeAndReturn() {
+      window.opener.location.replace("ViewPublication?PubId="+pubId);
+      window.close();
+    }
+
+    function displayPublications(CompoId,topicId){
+      var ieFix = new Date().getTime();
```
[the remark this answers is not in the document]
Comment (SilverDav) [reply kmelia/kmelia-war/src/main/webapp/kmelia/jsp/publicationLinksManager.jsp]: Done.

File: kmelia/kmelia-war/src/main/webapp/kmelia/jsp/publicationLinksManager.jsp [hunk 6]
```diff
@@ -26,169 +26,187 @@
...
+      $.get(getWebContext()+'/RAjaxPublicationsListServlet', {PubId:pubId,ComponentId:CompoId,TopicToLinkId:topicId,ToLink:1,Sort:currentSort,IEFix:ieFix},
+          function(data){
+            $('#pubList').html(data);
+          },"html");
+    }
+
+    function sendPubId(pubId,checked){
+
+      var action;
```
[the remark this answers is not in the document]
Comment (SilverDav) [reply kmelia/kmelia-war/src/main/webapp/kmelia/jsp/publicationLinksManager.jsp]: Done.

File: kmelia/kmelia-war/src/main/webapp/kmelia/jsp/publicationLinksManager.jsp [hunk 7]
```diff
@@ -26,169 +26,187 @@
...
+
+      if(checked){
+        action="Action=bindToPub";
+      }else{
+        action="Action=unbindToPub";
+      }
+      var ieFix = new Date().getTime();
```
[the remark this answers is not in the document]
Comment (SilverDav) [reply kmelia/kmelia-war/src/main/webapp/kmelia/jsp/publicationLinksManager.jsp]: Done.

**Review by mmoqui — APPROVED** [review]

**Pull request merged**